pkg/kubectl/cmd/util/factory.go

```diff
@@ -355,9 +405,9 @@ func (f *factory) DiscoveryClient() (discovery.CachedDiscoveryInterface, error)
 	return NewCachedDiscoveryClient(discoveryClient, cacheDir, time.Duration(10*time.Minute)), nil
 }
 
-func (f *factory) Object() (meta.RESTMapper, runtime.ObjectTyper) {
+func Object(dcf DiscoveryClientFactory, ccf ClientCacheFactory) (meta.RESTMapper, runtime.ObjectTyper) {
 	mapper := registered.RESTMapper()
-	discoveryClient, err := f.DiscoveryClient()
+	discoveryClient, err := dcf.DiscoveryClient()
```

Comment:
Needing this for a single release to support `/oapi` really seems more like "carry your patch for a release", than a "tear open the upstream code".  Especially since we're fixing downstream to use /apis and upstream to use `UnstructuredObject`